### PR TITLE
add: imagePullSecret support for private container registry

### PR DESCRIPTION
**Pull request by tylern91**
Signed-off-by: Tyler Nguyen <tyler.nguyen1191@gmail.com>

This PR adds support for `imagePullSecrets` argument, which is required for the use of mirrored image from the private container registry.

### CODE REVIEW

File: stable/coredns/templates/serviceaccount-autoscaler.yaml [hunk 1]
```diff
@@ -18,4 +18,9 @@ metadata:
 {{- if .Values.customLabels }}
 {{ toYaml .Values.customLabels | indent 4 }}
 {{- end }}
+{{- if .Values.autoscaler.image.pullSecrets }}
+imagePullSecrets:
+{{- range .Values.autoscaler.image.pullSecrets }}
+  - name: {{ . }}
+{{- end }}
```
Comment (mrueg) [inline stable/coredns/templates/serviceaccount-autoscaler.yaml:25]: Is this missing a second {{- end }}?

File: stable/coredns/templates/serviceaccount.yaml [hunk 2]
```diff
@@ -17,4 +17,9 @@ metadata:
   annotations:
     {{- toYaml . | nindent 4 }}
   {{- end }}
+{{- if .Values.image.pullSecrets }}
+imagePullSecrets:
+{{- range .Values.image.pullSecrets }}
+  - name: {{ . }}
+{{- end }}
```
Comment (mrueg) [inline stable/coredns/templates/serviceaccount.yaml:24]: Same here, missing a second end to close if

File: stable/coredns/templates/serviceaccount-autoscaler.yaml (hunk 1 above)
Comment (tylern91) [reply stable/coredns/templates/serviceaccount-autoscaler.yaml:25]: Fixed and rebased the commits (https://github.com/coredns/helm/pull/36/commits/1df87bc1e980491ac78e6d49a4f287039aa4b333)

File: stable/coredns/templates/serviceaccount.yaml (hunk 2 above)
Comment (tylern91) [reply stable/coredns/templates/serviceaccount.yaml:24]: Fixed and rebased the commits (https://github.com/coredns/helm/pull/36/commits/1df87bc1e980491ac78e6d49a4f287039aa4b333)

**Review by mrueg — APPROVED** [review]

**Pull request merged**